doc/specs/#980 - SnapOnOutput.md

```diff
@@ -0,0 +1,100 @@
+The default `snapOnOutput` value will be `[ "noSelection", "atBottom" ]`.
+
+When an enum array is defined in the settings, it will be interpreted logically. The following scenarios will be interpreted as follows:
+- `[ "always", "atBottom" ]` --> `"always"` because `atBottom` is a subset of `always`
+- `[ "never", "atBottom" ]` --> `"atBottom"`
+
+## Capabilities
+
+### Accessibility
+
+The `UiaRenderer` exposes new content that is rendered to the screen upon a scroll. No additional changes are necessary here.
+
+### Security
+
+N/A
+
+### Reliability
+
+N/A
+
+### Compatibility
+
+N/A
+
+### Performance, Power, and Efficiency
+
+N/A
+
+## Potential Issues
+
+### Circling the buffer
+If the text buffer fills up, the text buffer begins 'circling'. This means that new output shifts lines of the buffer up to make space. In a case like this, if `snapOnOutput` is set to `never`, the viewport should actually scroll up to keep the same content on the viewport.
+
+In the event that the buffer is circling and the viewport has been moved to the top of the buffer, that content of the buffer is now lost (as the 'Infinite Scrollback' feature does not exist or is disabled). At that point, the viewport will remain at the top of the buffer and the new output will push old output out of the buffer.
+
+### Infinite Scrollback
+See **Future considerations** > **Infinite Scrollback**.
+
+## Future considerations
+
+### Extensibility
+The introduction of `enum SnapOnOutput` allows for this feature to be enabled/disabled in more complex scenarios. A potential extension would be to introduce a new UI element or keybinding to toggle this feature.
+
+Another would be to introduce additional accepted values. Consider the output from a large testing suite. A user could create an extension that snaps on output for failing tests or when the testing has concluded.
+
+### Infinite Scrollback
+At the time of introducing this, the infinite scrollback feature is not supported. This means that the buffer saves the history up to the `historySize` amount of lines. When infinite scrollback is introduced, the buffer needs to change its own contents to allow the user to scroll beyond the `historySize`. With infinite scrollback enabled and the mutable viewport **NOT** snapping to new output, the `TerminalCore` needs to keep track of...
+- what contents are currently visible to the user (in the current location of the mutable viewport)
+- how to respond to a user's action of changing the location of the mutable viewport (i.e.: snapOnInput, scroll up/down)
+
+The `TermControl`'s scrollbar will abstract this issue away. As the viewport moves, really we're just presenting content that was saved to the text buffer.
+
+### Private Mode Escape Sequences
+There are a couple of private mode escape sequences that some terminals use to control this kind of thing. Mode 1010, for example, snaps the viewport to the bottom on output, whereas Mode 1011 spans the viewport to the bottom on a keypress.

```

Comment:
Please include citations.

Reply:
I'm not sure how stable the fragment parts are, but these are direct links to the modes on the _XTerm Control Sequences_ page:

* [DECSET 1010](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h4-Functions-using-CSI-_-ordered-by-the-final-character-lparen-s-rparen:CSI-?-Pm-h:Ps-=-1-0-1-0.1F79)
* [DECSET 1011](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h4-Functions-using-CSI-_-ordered-by-the-final-character-lparen-s-rparen:CSI-?-Pm-h:Ps-=-1-0-1-1.1F7A)

Otherwise you could probably just link to the root of the page.